packages/svelte/src/compiler/phases/2-analyze/visitors/ExpressionTag.js

```diff
@@ -15,9 +16,17 @@ export function ExpressionTag(node, context) {
 		}
 	}
 
-	// TODO ideally we wouldn't do this here, we'd just do it on encountering
-	// an `Identifier` within the tag. But we currently need to handle `{42}` etc
-	mark_subtree_dynamic(context.path);
+	const attribute_parent = context.path.find((parent) => parent.type === 'Attribute');
+	/**
+	 * if the expression tag is part of an attribute we want to check if it's inlinable before marking
+	 * the subtree as dynamic. This is because if it's inlinable it will be inlined in the template
+	 * directly making the whole thing actually static.
+	 */
+	if (!attribute_parent || !is_inlinable_expression(node, context.state.scope)) {
+		// TODO ideally we wouldn't do this here, we'd just do it on encountering
+		// an `Identifier` within the tag. But we currently need to handle `{42}` etc
+		mark_subtree_dynamic(context.path);
+	}
```

Comment:
per the 'TODO ideally we wouldn't do this here' comment, what would it take to get rid of this altogether and delegate to the `Identifier` visitor?

Reply:
i don't think we can remove the todo until we get rid of
```svelte
<div title={42}></div>
```
in svelte 6 (? i don't know if that's the plan) because this is a valid syntax, 42 is not an identifier and the subtree needs to be dynamic in that case because the generated code looks like this

```js
import "svelte/internal/disclose-version";
import "svelte/internal/flags/legacy";
import * as $ from "svelte/internal/client";

var root = $.template(`<div><div></div></div>`);

export default function App($$anchor) {
	var div = root();
	var div_1 = $.child(div);

	$.set_attribute(div_1, "title", 42);
	div_1.textContent = `Hello ${name ?? ""}!`;
	$.reset(div);
	$.append($$anchor, div);
}
```

Reply:
Surely we could emit this instead? Are there cases where inlining static values would fail?

```js
var root = $.template(`<div title="42"></div>`);

export default function App($$anchor) {
	var div = root();

	$.append($$anchor, div);
}
```

Reply:
(I realise this may be a distraction from the issue at hand, I'm just trying to avoid us adding _more_ duplicative code than is already there)

Reply:
> (I realise this may be a distraction from the issue at hand, I'm just trying to avoid us adding more duplicative code than is already there)

I mean if we can get the compiler output even better it's even better 😄  let me see

Reply:
So there are some tests failing, i think one of the main problems is that the original work to inline in the template made by @benmccann covered attributes but not expression tags in the dom. There's also some other kink but i guess it could be made work...do we want to do it in a follow up?

Reply:
I think we should prioritise it now, personally — the current approach just doesn't make a whole lot of sense to me if I'm honest

Reply:
Ok i'll need a bit more time then...i think i can make it work but the change is a bit more involved and touches a bit more files. Will continue working on this tomorrow probably.